packages/component-aws-s3/README.md

```diff
@@ -0,0 +1,213 @@
+
+## Configuring a DBOS Application with AWS S3
+First, ensure that the DBOS S3 component is installed into the application:
+```
+npm install --save @dbos-inc/component-aws-s3
+```
+
+Second, ensure that the library class is imported and exported from an application entrypoint source file:
+```typescript
+import { S3Ops } from "@dbos-inc/component-aws-s3";
+export { S3Ops };
+```
+
+Third, place appropriate configuration into the [`dbos-config.yaml`](https://docs.dbos.dev/api-reference/configuration) file; the following example will pull the AWS information from the environment:
+```yaml
+application:
+  aws_s3_configuration: aws_config # Optional if the section is called `aws_config`
+  aws_config:
+    aws_region: ${AWS_REGION}
+    aws_access_key_id: ${AWS_ACCESS_KEY_ID}
+    aws_secret_access_key: ${AWS_SECRET_ACCESS_KEY}
+```
+
+If a different configuration file section should be used for SES, the `aws_s3_configuration` can be changed to indicate a configuration section for use with SES.  If multiple configurations are to be used, the application code will have to name and configure them.
+
+The application will likely need an s3 bucket.  This can be placed in the `application` section of `dbos-config.yaml` also, but the naming key is to be established by the application.
+
+## Selecting A Configuration
+`S3Ops` is a configured class.  This means that the configuration (or config file key name) must be provided when a class instance is created, for example:
+```typescript
+const defaultS3 = configureInstance(S3Ops, 'default', {awscfgname: 'aws_config', bucket: 'my-s3-bucket', ...});
+```
+
+## Simple Operation Wrappers
+The `S3Ops` class provides several [communicator](https://docs.dbos.dev/tutorials/communicator-tutorial) wrappers for S3 functions.
+
+### Reading and Writing S3 From DBOS Handlers and Workflows
+
+#### Writing S3 Objects
+A string can be written to an S3 key with the following:
+```typescript
+    const putres = await ctx.invoke(defaultS3).putS3Comm('/my/test/key', "Test string from DBOS");
+```
+
+### Reading S3 Objects
+A string can be read from an S3 key with the following:
+```typescript
+    const getres = await ctx.invoke(defaultS3).getS3Comm('/my/test/key');
+```
+
+### Deleting Objects
+An S3 key can be removed/deleted with the following:
+```typescript
+    const delres = await ctx.invoke(defaultS3).deleteS3Comm('/my/test/key');
+```
+
+### Client Access To S3 Objects
+As shown below, DBOS provides a convenient and powerful way to track and manage S3 objects.  However, it is often not convenient to send or retrieve large S3 object contents through DBOS; it would be preferable to have the client talk directly to S3.  S3 accomodates this use case very well, using a feature called ["presigned URLs"](https://docs.aws.amazon.com/AmazonS3/latest/userguide/PresignedUrlUploadObject.html).
```

Comment:
"it is often not convenient to send or retrieve large S3 object contents through DBOS" --> I didn't notice this until I read through it again. Need to bring up the large file issue more prominently up front. For example, note that never directly pass in large file content through the input of a workflow or output of any DBOS functions, because they'd be recorded in the database.